plugins/node_block_sync/main.go

```diff
@@ -3,88 +3,98 @@ package main
-	defaultAddr = "127.0.0.1"
-	defaultPort = 9091
-
-	pluginName = "cosmos-sdk-block-height"
-
-	// Parameters for alert
-	maxHistoryCount = 100
-	fetchInterval   = time.Second * 2
-
-	estimateTimeWindow     = time.Second * 30
-	blockIntervalThreshold = time.Millisecond * 7500
+	defaultRPCAddr = "http://localhost:26657"
+	defaultAddr    = "127.0.0.1"
+	defaultPort    = 9091
+	pluginName     = "cosmoshub-block-sync"
+	defaultCriticalCount = 3
 )
 
 var (
-	addr string
-	port int
-
-	collector status.Collector
-	estimator policy.Estimator
+	rpcAddr		string
+	addr		string
+	port		int
+	prevHeight	int
+	latestHeight	int
+	warningCount	int
+	criticalCount   int
 )
 
 func init() {
-	collector = status.NewStatusCollector()
-	estimator = policy.NewEstimator()
-}
-
-func main() {
-	flag.StringVar(&addr, "addr", defaultAddr, "IP Address(e.g. 0.0.0.0, 127.0.0.1)")
-	flag.IntVar(&port, "port", defaultPort, "Port number, default 9091")
+	flag.StringVar(&rpcAddr, "rpcURI", defaultRPCAddr, "Tendermint RPC URI Address")
+	flag.StringVar(&addr, "addr", defaultAddr, "Listening address")
+	flag.IntVar(&port, "port", defaultPort, "Listening port")
+	flag.IntVar(&criticalCount, "critical", defaultCriticalCount, "block height stucked count to raise critical level of alert")
 
 	flag.Parse()
+}
 
-	collector.Start(maxHistoryCount, fetchInterval)
-
+func main() {
 	p := sdk.NewPlugin(pluginName)
 	p.Register(pluginFeature)
 
 	ctx := context.Background()
 	if err := p.Start(ctx, addr, port); err != nil {
-		// TODO: Stop collector
 		fmt.Println("exit")
 	}
 }
 
 func pluginFeature(info, option map[string]*structpb.Value) (sdk.CallResponse, error) {
-	log.Info().Str("main", "main").Msg(fmt.Sprintf("pluginFeature: %s", info["execute_method"]))
-
-	// TODO: filter out by execute_method.
-	histories := collector.GetHistories()
-	status, err := estimator.Estimate(histories, estimateTimeWindow, blockIntervalThreshold)
-	if err != nil {
-		log.Error().Str("main", "main").Msg(err.Error())
-		return sdk.CallResponse{}, err
+	severity := pluginpb.SEVERITY_INFO
+	state := pluginpb.STATE_NONE
+
+	var msg string
+
+	status, err := rpcCosmos.GetStatus(rpcAddr)
+
+	if err == nil {
+		latestHeight, _ = strconv.Atoi(status.Result.SyncInfo.LatestBlockHeight)
+		log.Info().Str("module", "plugin").Msgf("previous block height: %d, latest block height: %d", prevHeight, latestHeight)
+		state = pluginpb.STATE_SUCCESS
+
+		if latestHeight > prevHeight {
+			severity = pluginpb.SEVERITY_INFO
+			msg = fmt.Sprintf("block height increasing")
+			warningCount = 0
+		} else {
+			severity = pluginpb.SEVERITY_WARNING
+			warningCount++
+			msg = fmt.Sprintf("block height stucked %d times", warningCount)
+		}
+
+		if warningCount > criticalCount {
+			severity = pluginpb.SEVERITY_CRITICAL
+			msg = fmt.Sprintf("block height stucked more than %d times", warningCount)
+		}
+		log.Debug().Str("module", "plugin").Msg(msg)
+	} else {
+		// Maybe node wil be killed. So other alert comes to you.
```

Comment:
I think this need to fix typo :)
`wil be` -> `will be`

Reply:
Fixed. Thanks!
9a3a4015b602ac02d18e0144df876abcda89aedd